src/main/java/io/projectriff/invoker/server/GrpcServerAdapter.java

```diff
@@ -138,6 +146,47 @@ private OutputSignal fromSpringMessage(Tuple2<Integer, Message<byte[]>> out) {
                 ;
     }
 
+    // Used to transform the publisher chain into one that doesn't forward cancel() calls once it has complete()d.
+    private Function<? super Publisher<Tuple2<Integer, Message<byte[]>>>, ? extends Publisher<Tuple2<Integer, Message<byte[]>>>> ignoreCancelsAfterComplete() {
+        return Operators.lift((f, actual) ->
+                new CoreSubscriber<>() {
+                    AtomicBoolean completed = new AtomicBoolean();
```

Comment:
since we don't need atomic CAS operation in `onComplete`, you could also use `volatile boolean completed = false` instead of +1 allocating `AtomicBoolean`